src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4595,10 +4595,15 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
 
         val tp = tpt1.tpe
         val sym = tp.typeSymbol.initialize
-        if ((sym.isAbstractType || sym.hasAbstractFlag)
-            && !(sym.isJavaAnnotation && context.inAnnotation))
-          IsAbstractError(tree, sym)
-        else if (isPrimitiveValueClass(sym)) {
+
+        if (tpt.isErroneous) {
+          setError(tpt)
+        } else if (context.typingAnnotationCore && !sym.isJavaAnnotation && !(sym isNonBottomSubClass AnnotationClass)) {
```

Comment:
I don't really like having both `TypingAnnotation` and `TypingAnnotationCore` around. 

What if we checked this in `typedAnnotation`, after typing the core? Something like
```scala
val pt = (if (typedFun0.symbol.isJavaAnnotation) JavaAnnotationClass else AnnotationClass).tpe
if (!typedFun0.tpe <:< pt) NotAnAnnotationError(tree, typedFun0.symbol)
```
?

Reply:
So it seems like that would work for most cases except test/files/jvm/annotations/Test_2.scala:97

typedFun0 would be false for isJavaAnnotation but .baseClasses shows it extends java.lang.annotation. Not sure if that's a bug. I did something else. lmk what you think